lib/flipper/middleware/eager_memoizer.rb

```diff
@@ -0,0 +1,48 @@
+require 'rack/body_proxy'
+
+module Flipper
+  module Middleware
+    class EagerMemoizer
```

Comment:
I like the idea of making it into an option of `Memoizer` as it would avoid them being able to be registered in the wrong order and so on.